bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/async/HeapSizeManager.java

```diff
@@ -63,61 +73,74 @@ public long getMaxHeapSize() {
     return maxHeapSize;
   }
 
-  public synchronized void waitUntilAllOperationsAreDone() throws InterruptedException {
+  public synchronized void flush() throws InterruptedException {
+    isFlushing.set(true);
     boolean performedWarning = false;
-    while(!pendingOperationsWithSize.isEmpty()) {
+    while (true) {
       cleanupFinishedOperations();
       if (pendingOperationsWithSize.isEmpty()) {
         break;
       }
-      if (!performedWarning
-          && lastOperationChange + INTERVAL_NO_SUCCESS_WARNING < System.currentTimeMillis()) {
-        long lastUpdated = (System.currentTimeMillis() - lastOperationChange) / 1000;
-        LOG.warn("No operations completed within the last %d seconds."
-            + "There are still %d operations in progress.", lastUpdated,
-          pendingOperationsWithSize.size());
+      if (!performedWarning && requiresWarning()) {
+        warn();
         performedWarning = true;
-        waitForCompletions(FINISH_WAIT_MILLIS);
       }
+      wait(FINISH_WAIT_MILLIS);
     }
     if (performedWarning) {
```

Comment:
If performedWarning is false, we don't necessarily want to log that flush() is done.  We only want to performedWarning if something seems wrong.  This follow up of "flush() complete" will show that the situation corrected itself.  We usually don't actually want to LOG.info() when flush completes... we only want to do it in unusual circumstances.